internal/mage/engine.go

```diff
@@ -73,21 +73,32 @@ if [ "$` + servicesDNSEnvName + `" != "0" ]; then
 	grep -v '^nameserver' /etc/resolv.conf.upstream >> /etc/resolv.conf
 fi
 
-exec {{.EngineBin}} --debug --config {{.EngineConfig}} --oci-worker-binary {{.ShimBin}}
+exec {{.EngineBin}} --config {{.EngineConfig}}
+`
+
+var engineConfig string
+
+const engineConfigTmpl = `
+debug = true
+insecure-entitlements = ["security.insecure"]
+
+[worker.oci]
+  enabled = true
+	binary = "{{.ShimBin}}"
+	[worker.oci.labels]
+	    "privilegedEnabled" = "true"
```

Comment:
Previously I switched to having a blank default `engine.toml` and applying our own defaults at runtime, to make this config additive:

https://github.com/dagger/dagger/blob/275a7135e4f95f0ba1d13a7f48cdfb467809b1e7/cmd/engine/config.go#L32

The advantage of doing it this way is that when someone needs to add e.g. registry mirror config to `engine.toml` they don't need to keep updating it to include whatever other defaults we add.

`insecure-entitlements` is a little tricky though because it's hard to differentiate between the Buildkit default empty value and the user intentionally setting it to empty to prevent our default. :thinking: 

One option is to add our own flag to the engine like `--disable-insecure-root-capabilities` that prevents us from adding these defaults.

Reply:
Yep that makes sense, what do you think about this:
1. Update the setting of the worker label to be automatic in main based on whether the entitlement is present or not. This remove the need for that whole section of config.
2. Move the default shim binary setting to also be in our main, next to the other default engine config settings you added
3. Leave in a a default toml config, but now it consists just of
```
debug = true
insecure-entitlements = ["security.insecure"]
```

I updated with the above, let me know what you think. My concern with `--disable-insecure-root-capabilities` as a flag is that it gets confusing to explain in combination with the existing buildkit flag and in general I think it's simpler to only have to explain to users where they need to change a toml config file rather than start explaining mixes of toml configs, command line flags, entrypoint settings, etc.

Users who want a custom toml config will need to include those two lines, but that seems like not too much to ask or subject to lots of change in the same way the more complicated networking/worker configs are. So overall felt like a good balance.

Reply:
Nice, yeah that seems like a good balance since now `engine.toml` only contains optional things and an empty one is valid again. 👍